### PR TITLE
Change union to not remove all null values

### PR DESCRIPTION
**Pull request by mayerm94**
https://jira.mitre.org/browse/BONNIE-1371
Do distinct will now not discard null values.

Pull requests into cql-execution require the following.
Submitter and reviewer should ✔ when done.
For items that are not-applicable, mark "N/A" and ✔.

[CDS Connect](https://cds.ahrq.gov/cdsconnect) and [Bonnie](https://github.com/projecttacoma/bonnie) are the main users of this repository. 
It is strongly recommended to include a person from each of those projects as a reviewer.

**Submitter:**
- [x] This pull request describes why these changes were made
- [x] Code diff has been done and been reviewed (it does not contain: additional white space, not applicable code changes, debug statements, etc.)
- [x] Tests are included and test edge cases
- [x] Tests have been run locally and pass
- [x] Code coverage has not gone down and all code touched or added is covered.
- [x] All dependent libraries are appropriately updated or have a corresponding PR related to this change
- [x] Regression test passes

Expected Regression Between Bonnie 2.1 and This in cql_qdm_patientapi
hmdmlantana
====Patients Passing Regression===========
+: Population: Stratification 3  SECOND BRANCH HAS MORE PASSES (24 < 33)
Measures with no differences: 14/15
====Code Coverage Percentage Regression===========
+: CMS ID: CMS111v7-0 SECOND BRANCH HAS MORE COVERAGE (98 < 100)
+: CMS ID: CMS55v7-0 SECOND BRANCH HAS MORE COVERAGE (98 < 100)
+: CMS ID: CMS32v8-0-1-2 SECOND BRANCH HAS MORE COVERAGE (82 < 100)
Measures with no differences: 9/12

**Reviewer:**

Name:
- [ ] Code is maintainable and reusable, reuses existing code and infrastructure where appropriate, and accomplishes the task’s purpose
- [x] The tests appropriately test the new code, including edge cases
- [x] You have tried to break the code

### CODE REVIEW

File: test/elm/list/test.coffee [hunk 1]
```diff
@@ -87,6 +87,9 @@ describe 'Union', ->
 
   it 'should remove duplicate elements (according to CQL 1.2 spec)', ->
     @oneToFiveOverlapped.exec(@ctx).should.eql [1, 2, 3, 4, 5]
+
```
Comment (losborne) [inline test/elm/list/test.coffee]: There is some whitespace at the beginning of this line that should be deleted.

File: test/elm/list/test.coffee [hunk 2]
```diff
@@ -384,6 +387,9 @@ describe 'Distinct', ->
 
   it 'should do nothing to an array of distinct tuples', ->
     @noDupsTuples.exec(@ctx).should.eql [{ hello: 'world' }, { hello: 'cleveland' }]
+
```
Comment (losborne) [inline test/elm/list/test.coffee]: There is some whitespace at the beginning of this line that should be deleted.

File: src/util/comparison.coffee [hunk 3]
```diff
@@ -45,6 +45,8 @@ module.exports.greaterThanOrEquals = (a, b, precision = DateTime.Unit.MILLISECON
 module.exports.equivalent = equivalent = (a, b) ->
   if a == null && b == null
     return true
+  else if a == null || b == null
```
Comment (losborne) [inline src/util/comparison.coffee]: `a == null` will get transpiled to `a === null` in JavaScript, which will fail for `undefined` input.  What we really want is `a == null` in Javascript which is `a?` in CoffeeScript.

Lines 46-49 on this branch should go away when the merge conflict with https://github.com/cqframework/cql-execution/pull/9 gets resolved.  It might be best to replace them here just to reduce the risk that these lines make it through instead somehow.

They should instead be:
```
return true if not a? and not b?
return false unless a? and b?
```
